crates/core/src/host/instance_env.rs

```diff
@@ -194,13 +194,13 @@ impl InstanceEnv {
     ///
     /// Errors with `TableNotFound` if the table does not exist.
     #[tracing::instrument(skip_all)]
-    pub fn get_table_id(&self, table_name: String) -> Result<TableId, NodesError> {
+    pub fn get_table_id(&self, table_name: &str) -> Result<TableId, NodesError> {
         let stdb = &*self.dbic.relational_db;
         let tx = &mut *self.get_tx()?;
 
         // Query the table id from the name.
         let table_id = stdb
-            .table_id_from_name(tx, &table_name)?
+            .table_id_from_name(tx, table_name)?
```

Comment:
This seems like a solid change, but appears not to be relevant to this PR.

Reply:
It is relevant, since with wasmtime we actually do get a `str` reference directly to memory instead of being forced to clone it out, so without this change there'd be a new explicit `.to_owned()` in wasm_instance_env. If you do feel strongly about it I can do that and leave this as a follow-up?

Reply:
Nah, as long as it's connected to this PR I'm glad to have it in. Just trying to be diligent about not merging in unrelated changes.